### PR TITLE
Refactor AwsFog and ProvisionWorker from D

### PR DESCRIPTION
**Pull request by calebthompson**
https://codeclimate.com/repos/54d4fa64695680762a0024c5/ProvisionWorker
https://codeclimate.com/repos/54d4fa64695680762a0024c5/AwsFog

### CODE REVIEW

**Comment by chriskacerguis** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/54d4fa64695680762a0024c5/compare/refactor-ds">analyzed this pull request</a>.

**Comment by calebthompson** [comment]
Ping @jntullo since you'd touched these most recently.

**Comment by calebthompson** [comment]
Resolves #298 and #299

**Comment by jntullo** [comment]
@calebthompson @chriskacerguis currently resolving issues with provisioning and adding some more refactoring.

File: app/workers/aws_fog.rb [hunk 1]
```diff
@@ -5,18 +5,20 @@ def initialize(order_item_id)
...
-    rescue Excon::Errors::BadRequest
-      order_item.provision_status = :critical
-      order_item.status_msg = 'Bad request. Check authorization credentials.'
-    rescue ArgumentError, StandardError, Fog::Compute::AWS::Error, NoMethodError  => e
-      order_item.provision_status = :critical
-      order_item.status_msg = e.message
-    ensure
-      order_item.save
-    end
+    product_provisioner.provision(order_item, aws_settings)
```
Comment (jntullo) [inline app/workers/aws_fog.rb]: When running this, getting an uninitialized constant. product_provisioner is not working properly with the new classes.

**Comment by chriskacerguis** [comment]
This causes provisioning to break, and is a dup ticket of #298 #299

**Comment by chriskacerguis** [comment]
Closing pull request.  
No merge.
:-1:

**Pull request closed without merging**